### PR TITLE
Fix redirect security

### LINKED ISSUE #720: Return link without validations is potentially unsafe
The feature introduced in #674 is problematic. This was noticed by @luniki. Some article about this security concern: https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html

One solution would be to have a whitelist of domains that can be redirected to. That whitelist would only be configurable via `settings.toml` and NOT via GET parameters. This would also mean that other apps already using `return.target` and `return.label` don't have to change.

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Fixes #720 

CC @luniki

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -149,13 +149,22 @@ parameter as well. See further below for information on that.
 # right above the "start a new recording" button. This button links to the
 # value of this key. Setting this value is usually done via GET parameters, as
 # a global configuration often does not make sense. You also probably want to
-# set `return.label`.
+# set `return.label`. Also see `return.allowedDomains`.
 #target = "https://my-lsm.me.com/course?id=123"
```
Comment (lkiesow) [inline CONFIGURATION.md:153]:
```suggestion
#target = "https://my-lms.me.com/course?id=123"
```

File: CONFIGURATION.md [hunk 2]
```diff
@@ -149,13 +149,22 @@ parameter as well. See further below for information on that.
...
 
 # Label for the site that `return.target` links to. Usually the name of your
 # LMS. The English button label is "Exit and go back to {{return.label}}". This
 # value is only used if `return.target` is set!
 #label = "Stud.IP"
+
+# A list of domains that are allowed for `return.target` links. The domain on
+# which Studio is served is always allowed. This is for security, as
+# redirecting to URLs that could be set by an attacker is a bad idea. See here:
+# https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html
```
Comment (lkiesow) [inline CONFIGURATION.md:163]: The second part is not really relevant for someone configuring this and long tests just lead to people not reading this.

```suggestion
# which Studio is served is always allowed.
```

**Review by lkiesow — COMMENTED** [review]
Just some minor things…

File: src/settings.js [hunk 3]
```diff
@@ -238,10 +252,10 @@ export class SettingsManager {
   }
 
   // Validate the given `obj` with the global settings `SCHEMA`. If `allowParse`
-  // is true, string values are attempted to parse into the expected type.
-  // `source` is just a string for error messages specifying where `obj` comes
-  // from.
-  validate(obj, allowParse, source) {
+  // is true, string values are attempted to parse into the expected type. `src`
+  // must be one of `SRC_SERVER` and `SRC_URL`. `srcDescription` is just a
```
Comment (JulianKniephoff) [inline src/settings.js:256]: What about `SRC_LOCAL_STORAGE`?

**Review by JulianKniephoff — COMMENTED** [review]
Only a minor nitpick about a docstring.

**Review by JulianKniephoff — APPROVED** [review]

**Pull request merged**